### PR TITLE
Updated: Add read/write and grant config info for external etcd in cluster guide

### LINKED ISSUE #2184: Sensu Go + etcd authentication (grant-permission)
As an engineer, 
I would like to ensure that my `backend-etcd` user has the correct permissions when `etcd` authentication is enabled.

## Feature Suggestion
Documentation on how to run Sensu Go while utilizing `etcd` authentication.  This will further enhance how to run Sensu Go securely.

## Possible Implementation
Add documentation to the follow page + h2
https://docs.sensu.io/sensu-go/latest/guides/securing-sensu/#etcd-user-role

## Context
Using etcd 3.4 with passwordless user account via mTLS with authentication enabled, I was unable to get Sensu Go `backend init` to run properly.

```
{"level":"warn","ts":"2020-02-13T15:31:24.552-0600","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-424eebe0-8294-4eb0-b146-7c456157de7c/REDACTED:2379","attempt":0,"error":"rpc error: code = PermissionDenied desc = etcdserver: permission denied"}
```

The concerning part is that I had scoped quite a bit into the user/role in etcd.

```
[root@REDACTED ~]# /opt/etcd/etcdctl user get REDACTED --detail
User: REDACTED

Role sensu_readwrite
KV Read:
	/
	[/, 0) (prefix /)
	[/sensu.io, /sensu.ip) (prefix /sensu.io)
	[/sensu.io/, /sensu.io0) (prefix /sensu.io/)
KV Write:
	/
	[/, 0) (prefix /)
	[/sensu.io, /sensu.ip) (prefix /sensu.io)
	[/sensu.io/, /sensu.io0) (prefix /sensu.io/)
```

As a test, I disabled etcd authentication and it was able to initialize the backend.  This means that this is definitely related to Sensu Go + permissions needed in etcd.  

I will continue working on getting them working in my environment and contribute as I can.

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds information about configuring read/write access and grant permission for etcd auth.
- Adds etcd compatibility info to datastore reference.
- Updates etcd links to use 3.3.13 docs.
- Propagates changes to all currently published docs versions.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2184

### CODE REVIEW

File: content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md [hunk 1]
```diff
@@ -262,11 +262,44 @@ See [Secure Sensu][16] for information about cluster security.
...
+
+{{< highlight shell >}}
+# Add the sensu user (you will receive an interactive prompt for the password)
+etcdctl user add sensu
+# Create a role
+etcdctl role add sensu_readwrite
+# Give that role permission to read and write under the /sensu.io/ keyspace
+etcdctl role grant-permission sensu_readwrite readwrite --from-key '/sensu.io/'
+# Grant the sensu user the role
+etcdctl user grant-role sensu sensu_readwrite
```
Comment (hillaryfraley) [inline content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md:278]: @tarcinil your comment at https://github.com/sensu/sensu-docs/pull/2218#discussion_r435674198 indicates that we still need to add something more here, but I'm not sure exactly what. Can you clarify?

Comment (cwjohnston) [reply content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md:278]: I believe the comment is referring to the key at `/sensu.io/.initialized` which Sensu must be able to create in order to successfully initialize etcd database. AFAICT the current grant-permission would be sufficient to allow writing the `/sensu.io/.initialized` key.

Comment (cwjohnston) [reply content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md:278]: My tests are showing that @tarcinil's comment is correct -- an additional permission is needed for `sensu-backend init` to succeed. 

The init process tries to obtain a mutex lock, which attempts to create keys under `.initialized.lock`, so the permissions limited to `/sensu.io/` prefix are insufficient:

```
etcdserver: failed to apply request,took 16.179µs,request header:<ID:4554675860814564631 username:"sensu-agent" auth_revision:8 > txn:<compare:<target:CREATE key:".initialized.lock/3f3575904ee60d15" create_revision:0 > success:<request_put:<key:".initialized.lock/3f3575904ee60d15" lease:4554675860814564629 > request_range:<key:".initialized.lock/" range_end:".initialized.lock0" limit:1 sort_order:ASCEND sort_target:CREATE > > failure:<request_range:<key:".initialized.lock/3f3575904ee60d15" >  request_range:<key:".initialized.lock/" range_end:".initialized.lock0" limit:1 sort_order:ASCEND sort_target:CREATE > >>,resp ,err is auth: permission denied
```

I have tried granting permissions such as 
```
etcdctl role grant-permission sensu_readwrite readwrite --from-key '.initialized.lock'
```

But these too seem insufficient:

```
etcdserver: failed to apply request,took 9.573µs,request header:<ID:4554675860814564705 username:"sensu-agent" auth_revision:20 > txn:<compare:<target:CREATE key:".initialized.lock/3f3575904ee60d5f" create_revision:0 > success:<request_put:<key:".initialized.lock/3f3575904ee60d5f" lease:4554675860814564703 > request_range:<key:".initialized.lock/" range_end:".initialized.lock0" limit:1 sort_order:ASCEND sort_target:CREATE > > failure:<request_range:<key:".initialized.lock/3f3575904ee60d5f" >  request_range:<key:".initialized.lock/" range_end:".initialized.lock0" limit:1 sort_order:ASCEND sort_target:CREATE > >>,resp ,err is auth: permission denied
```

Using a slightly different key got a weird result:

```
etcdctl --user root:tacosrule role grant-permission sensu_readwrite readwrite --from-key '.initialized.lock/'
```

With this permission, sensu-backend init fails claiming permission denied, but the etcd server does not print any corresponding permission errors as it does with the other permissions tested. 🤔

File: content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md [hunk 2]
```diff
@@ -262,11 +262,44 @@ See [Secure Sensu][16] for information about cluster security.
 
 ## Use an external etcd cluster
 
-To use Sensu with an external etcd cluster, you must have etcd 3.3.2 or newer.
-To stand up an external etcd cluster, follow etcd's [clustering guide][2] using the same store configuration.
+Using Sensu with an external etcd cluster requires etcd 3.3.2 or newer but is not compatible with etcd 3.4.0 or newer.
```
Comment (amdprophet) [inline content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md:265]: Sensu is compatible with Etcd 3.4.0 or greater however we've seen a ~50% decrease in performance when using Etcd-only configurations. We may want to specify that it's compatible but slower than Etcd 3.3.

Comment (tarcinil) [reply content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md:265]: This statement makes so many things make sense now.

Comment (rpunt) [reply content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md:265]: I really REALLY wish the 3.4.x performance callout had been present when we migrated from sensu-core in 4 environments. :(

Comment (hillaryfraley) [reply content/sensu-go/5.20/operations/deploy-sensu/cluster-sensu.md:265]: I corrected this in https://github.com/sensu/sensu-docs/pull/3019. Thank you @amdprophet

**Comment by hillaryfraley** [comment]
Closed in favor of https://github.com/sensu/sensu-docs/pull/3130 to resolve conflicts.

**Pull request closed without merging**